libraries/AP_Mount/AP_Mount_Tusuav.cpp

```diff
@@ -0,0 +1,1109 @@
+                        (unsigned char)_parsed_msg.msg.data[6],
+                        (unsigned char)_parsed_msg.msg.data[7]); 
+                            
+                GCS_SEND_TEXT(MAV_SEVERITY_INFO, "%s BootVersion: %d.%d.%d %02d-%02d-%02d %02d:%02d", send_text_prefix,                                    
+                        (unsigned char)_parsed_msg.msg.data[8],
+                        (unsigned char)_parsed_msg.msg.data[9],
+                        (unsigned char)_parsed_msg.msg.data[10],
+                        (unsigned char)_parsed_msg.msg.data[11],
+                        (unsigned char)_parsed_msg.msg.data[12],
+                        (unsigned char)_parsed_msg.msg.data[13],
+                        (unsigned char)_parsed_msg.msg.data[14], 
+                        (unsigned char)_parsed_msg.msg.data[15]);
+                    break;
+            }
+            break;
+        }
+        default:
+            break;
+        }
+        break;
+    }
+
+    case GimbalCmd::MasCmd_Load:{
+
+        break;
+    }
+
+    default:
+        break;
+    }
+}
+
+// send packet to gimbal.  total_len includes everything including the header '0x5A' and crc
+// returns true on success, false if outgoing serial buffer is full or other errors
+bool AP_Mount_Tusuav::send_packet(struct FrameType frame)
+{
+    if (!_initialised) {
+        return false;
+    }
+
+    if (frame.data == NULL && frame.data_len > 0) {
+        return false;
+    }
+
+    frame.total_len = frame.data_len + AP_MOUNT_TUSUAV_PACKETLEN_MIN;
+    // calculate and sanity check packet size
+    if (frame.total_len >= AP_MOUNT_TUSUAV_PACKETLEN_MAX) {
+        debug("send_packet data buff too large");
+        return false;
+    }
+
+    // check for sufficient space in outgoing buffer
+    if (_uart->txspace() < frame.total_len) {
+        debug("tx space too low (%u < %u)", (unsigned)_uart->txspace(), (unsigned)frame.total_len);
+        return false;
+    }
+
+    // buffer for holding outgoing packet
+    uint8_t send_buff[frame.total_len];
+    uint8_t send_buff_ofs = 0;
+
+    // packet header length cmd and sub-cmd
+    send_buff[send_buff_ofs++] = AP_MOUNT_TUSUAV_HEADER1;
+    send_buff[send_buff_ofs++] = frame.total_len;
+    send_buff[send_buff_ofs++] = frame.flag.all;
+    send_buff[send_buff_ofs++] = frame.cmd;
+    send_buff[send_buff_ofs++] = frame.sub_cmd;
+
+    // data
+    if (frame.data_len != 0) {
+        memcpy(&send_buff[send_buff_ofs], frame.data, frame.data_len);
+        send_buff_ofs += frame.data_len;
+    }
+
+    // crc
+    send_buff[send_buff_ofs] = crc_calculate(send_buff, send_buff_ofs);
+    send_buff_ofs++;
+    // write packet to serial port
+    _uart->write(send_buff, send_buff_ofs);   
+
+    return true;
+}
+
+// send handshake, gimbal will respond with T1_F1_B1_D1 paket that includes current angles
+void AP_Mount_Tusuav::send_time_sync()
+{
+    // get current location
+    Location loc;
+    int32_t alt_amsl_cm = 0;
```

Comment:
```suggestion
    int32_t alt_amsl_cm;
```

Reply:
Why does using int32_t alt_amsl_cm not pass the check?

Reply:
Why does using int32_t alt_amsl_cm not pass the check? We compiled it properly in vscode for Ubuntu, but it didn't pass check.

Reply:
in vscode for Ubuntu：[1214/1215] Generating bin/arducopter.abin
[1215/1215] Generating bin/arducopter_with_bl.hex
githash 92693e023793133e49a035daf37c14433e484778 md5 0b7d876906442d26c6b3e18b746f0a2f
Created bin/arducopter.abin

Waf: Leaving directory `/home/sunyu/Ardupilot_Tusuav/build/CubeOrangePlus'

BUILD SUMMARY
Build directory: /home/sunyu/Ardupilot_Tusuav/build/CubeOrangePlus
Target          Text (B)  Data (B)  BSS (B)  Total Flash Used (B)  Free Flash (B)  External Flash Used (B)
----------------------------------------------------------------------------------------------------------
bin/arducopter   1952968      4192   258104               1957160            8916  Not Applicable         

'copter' finished successfully (8.565s)

Reply:
in vscode for Ubuntu：[1214/1215] Generating bin/arducopter.abin
[1215/1215] Generating bin/arducopter_with_bl.hex
githash 92693e0 md5 0b7d876906442d26c6b3e18b746f0a2f
Created bin/arducopter.abin

Waf: Leaving directory `/home/sunyu/Ardupilot_Tusuav/build/CubeOrangePlus'

BUILD SUMMARY
Build directory: /home/sunyu/Ardupilot_Tusuav/build/CubeOrangePlus
Target Text (B) Data (B) BSS (B) Total Flash Used (B) Free Flash (B) External Flash Used (B)
bin/arducopter 1952968 4192 258104 1957160 8916 Not Applicable

'copter' finished successfully (8.565s)

Reply:
in check：../../libraries/AP_Mount/AP_Mount_Tusuav.cpp:694:24: note: initialize the variable 'alt_amsl_cm' to silence this warning
    int32_t alt_amsl_cm;
                       ^
                        = 0
1 warning and 1 error generated.
Build failed
Waf: Leaving directory /Users/runner/work/ardupilot/ardupilot/build/sitl'
 -> task in 'objs/AP_Mount' failed (exit status 1): 
        {task 4394918160: cxx AP_Mount_Tusuav.cpp -> AP_Mount_Tusuav.cpp.0.o}
 (run with -v to display more information)
Error: Process completed with exit code 1.

Reply:
*sigh*.  The compiler is talking garbage here, but let's just leave the initialisation in.